### PR TITLE
Add TensorFlow checkpoint conversion tooling

### PR DESCRIPTION
**Pull request by kyhavlov**
Added the `scripts/convert_tf_checkpoint_to_jax.py` script which uses the toy dataset trajectories to compare logits/KL (using `all_d21_imitation_v3`):
```
mean_kl_tf_to_jax: mean=1.44475e-06 max=1.44475e-06
max_kl_tf_to_jax: mean=3.63034e-05 max=3.63034e-05
mean_kl_jax_to_tf: mean=1.49884e-06 max=1.49884e-06
max_abs_logit_diff: mean=0.119781 max=0.119781
```

### CODE REVIEW

File: scripts/convert_tf_checkpoint_to_jax.py [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+        with_nana=False,
+        legacy_jumps_left=True,
+    )
+    config['version'] = 5
+
+  if config.get('version') != 5:
+    raise ValueError(f"Unsupported legacy TF checkpoint version: {config.get('version')}")
+
+  controller = config['embed']['controller']
+  if 'type' not in controller:
```
Comment (vladfi1) [inline scripts/convert_tf_checkpoint_to_jax.py]: This should always be true.

File: scripts/convert_tf_checkpoint_to_jax.py [hunk 2]
```diff
@@ -0,0 +1,403 @@
...
+  if config.get('version') == 3:
+    config['observation'] = dataclasses.asdict(
+        observations.NULL_OBSERVATION_CONFIG)
+    config['version'] = 4
+
+  if config.get('version') == 4:
+    config['embed'].update(
+        with_randall=False,
+        with_fod=False,
+        items={'type': embed.ItemsType.SKIP.value},
```
Comment (vladfi1) [inline scripts/convert_tf_checkpoint_to_jax.py]: Ooc why not use the actual enum instead of its string value? Would eliminate the extra cast in embed.py.

**Review by vladfi1 — APPROVED** [review]
Looks good! Maybe just add a TODO for restoring optimizer state as well?

File: scripts/convert_tf_checkpoint_to_jax.py [hunk 3]
```diff
@@ -0,0 +1,403 @@
...
+import numpy as np
+from flax import nnx
+import tree
+
+from slippi_ai import data, observations, paths, reward, utils
+from slippi_ai.flag_utils import dataclass_from_dict
+from slippi_ai.jax import controller_heads, embed, jax_utils, networks, policies, saving
+
+
+ArrayTree = dict[str, tp.Any]
```
Comment (vladfi1) [inline scripts/convert_tf_checkpoint_to_jax.py]: Is the key actually `str | int`?

File: scripts/convert_tf_checkpoint_to_jax.py [hunk 4]
```diff
@@ -0,0 +1,403 @@
...
+
+
+def main():
+  parser = argparse.ArgumentParser(description=__doc__)
+  parser.add_argument('source', help='legacy TensorFlow pickle checkpoint')
+  parser.add_argument('output', help='output JAX pickle checkpoint')
+  parser.add_argument(
+      '--verify-data-dir',
+      default=str(paths.TOY_DATA_DIR),
+      help='run post-conversion TF/JAX parity on parsed game tables')
```
Comment (vladfi1) [inline scripts/convert_tf_checkpoint_to_jax.py]:
```suggestion
      help='run post-conversion TF/JAX parity on parsed game tables (defaults to toy data)')
```
